shared/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -245,14 +251,16 @@ namespace datadog::shared::nativeloader
             Log::Warn("CorProfiler::Initialize: Failed to attach profiler, interface ICorProfilerInfo4 not found.");
             // we're not recording the exact version here, we just know that at this point it's not enough
             single_step_guard_rails.RecordBootstrapError(SingleStepGuardRails::NetFrameworkRuntime, inferredVersion, "incompatible_runtime");
-            return E_FAIL;
+            m_disabled = true;
+            return S_OK;
```

Comment:
We only spawn the thread _if_ we're in an SSI scenario. Do we want to return `E_FAIL` in the situation where we are _not_ in SSI? 

i.e. should `single_step_guard_rails.RecordBootstrapError()` return a boolean (indicating whether we should return `S_OK/E_FAIL`)?